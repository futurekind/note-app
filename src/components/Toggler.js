import React, {PropTypes} from 'react';
import styled from 'styled-components';

import SectionTitle from './Sectiontitle';

const View = styled.aside``;

const Header = styled.header``;

const Body = styled.main``;

const Toggler = ({
    title,
    open,
    children,
    onToggle
}) => {
    return (
        <View>
            
            <Header onClick={ () => onToggle(open) }>
                <SectionTitle iconId={ open ? 'up' : 'down' }>{ title }</SectionTitle>
            </Header>

            <Body open={ open }>
                { children }
            </Body>
        </View>
    )
}

Toggler.defaultProps = {
    open: false,
    onToggle: () => {}
}

Toggler.propTypes = {
    title: PropTypes.string.isRequired,
    open: PropTypes.bool,
    onToggle: PropTypes.func,
    children: PropTypes.any,
}

export default Toggler