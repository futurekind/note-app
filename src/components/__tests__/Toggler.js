import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Toggler from '../Toggler';

describe('Toggler', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <Toggler title="Some Title" />
        )
    })

    it('renders', () => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('show up icon when open', () => {
        wrapper.setProps({
            open: true
        })

        const sectiontitle = wrapper.find('Sectiontitle')

        expect(toJSON(sectiontitle)).toMatchSnapshot()
    })

    it('renders children', () => {
        const wrapper = shallow(
            <Toggler title="title">
                <p>one</p>
                <p>two</p>
                <p>three</p>
            </Toggler>
        )

        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    it('delegates onToggle with false', () => {
        const spy = jest.fn();

        wrapper.setProps({
            onToggle: spy
        })

        const header = wrapper.children().at(0);
        
        header.simulate('click');

        expect(spy).toHaveBeenLastCalledWith(false)
    })

    it('delegates onToggle with true', () => {
        const spy = jest.fn();

        wrapper.setProps({
            onToggle: spy,
            open: true
        })

        const header = wrapper.children().at(0);

        header.simulate('click');

        expect(spy).toHaveBeenLastCalledWith(true)
    })

})