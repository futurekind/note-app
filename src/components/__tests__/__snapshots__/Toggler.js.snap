exports[`Toggler renders 1`] = `
<styled.aside>
  <styled.header
    onClick={[Function]}>
    <Sectiontitle
      iconId="down">
      Some Title
    </Sectiontitle>
  </styled.header>
  <styled.main
    open={false} />
</styled.aside>
`;

exports[`Toggler renders children 1`] = `
<styled.aside>
  <styled.header
    onClick={[Function]}>
    <Sectiontitle
      iconId="down">
      title
    </Sectiontitle>
  </styled.header>
  <styled.main
    open={false}>
    <p>
      one
    </p>
    <p>
      two
    </p>
    <p>
      three
    </p>
  </styled.main>
</styled.aside>
`;

exports[`Toggler show up icon when open 1`] = `
<Sectiontitle
  iconId="up">
  Some Title
</Sectiontitle>
`;
